src/core/model/Stroke.cpp

```diff
@@ -302,6 +302,22 @@ Point Stroke::getPoint(PathParameter parameter) const {
 
 auto Stroke::getPoints() const -> const Point* { return this->points.data(); }
 
+void Stroke::setPointVector(const std::vector<Point>& other, const Range* const snappingBox) {
+    this->points = other;
+    if (!snappingBox || this->points.empty() || this->points.front().z != Point::NO_PRESSURE) {
+        // We cannot deduce the bounding box from the snapping box if the stroke has pressure values
+        this->sizeCalculated = false;
+    } else {
+        assert(snappingBox->isValid());
```

Comment:
This assertion is not satisfied for a horizontal or vertical straight line (drawn with the "Line" drawing type, with grid snapping on) and the app crashes.

Reply:
Thanks for the catch! I'll get back to this PR in a moment.